lib/_http_client.js

```diff
@@ -57,8 +57,10 @@ function ClientRequest(options, cb) {
   const defaultPort = options.defaultPort ||
                       self.agent && self.agent.defaultPort;
 
-  var port = options.port = options.port || defaultPort || 80;
-  var host = options.host = options.hostname || options.host || 'localhost';
+  const hostParts = (options.host && options.host.split(':')) || [];
```

Comment:
this will fail on IPv6 address and port: `[::]:80`. Should probably use `url` to split here.